include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -409,7 +418,63 @@ class CBlitImageFilter : public CImageFilter<CBlitImageFilter<Normalize,Clamp,Sw
 					if (otherScale)
 					for (auto k=0; k<MaxChannels; k++)
 						scale.factor[k] *= otherScale->factor[k];
-						
+
+					// compute the phaseSupportLUT
+
+					// a dummy load functor
+					// does nothing but fills up the `windowSample` with 1s (identity) so we can preserve the value of kernel
+					// weights when eventually `windowSample` gets multiplied by them later in
+					// `CFloatingPointSeparableImageFilterKernelBase<CRTP>::sample_functor_t<PreFilter,PostFilter>::operator()`
+					// this exists only because `evaluateImpl` expects a pre filtering step.
+					// 
+					// Here I suggest to make something like a CImageFilterKernel<CRTP,value_type>::computeWeights which goes through
+					// the same pipeline of injecting other pre filtering stages (like `wrap` for the `CScaledImageFilterKernel`, for example)
+					// and which doesn't take this functor as argument.
+
+					auto dummyLoad = [](value_type* windowSample, const core::vectorSIMDf& unused0, const core::vectorSIMDi32& unused1,
+						const IImageFilterKernel::UserData* unused2) -> void
+					{
+						for (auto h = 0; h < MaxChannels; h++)
+							windowSample[h] = value_type(1);
+					};
+
+					// There's no real reason that this should have `MaxChannels` number of elements except for the fact that
+					// `CFloatingPointSeparableImageFilterKernelBase<CRTP>::sample_functor_t<PreFilter,PostFilter>::operator()` expects that.
+					// So this, for now, just replicates the same value in all 4 channels.
+					value_type kernelWeight[MaxChannels];
```

Comment:
there are special kernel wrappers which make them behave differently for different channels, the derivative map creator uses a blit filter with a separate-per-channel-kernel

Reply:
in the end `evaluateImpl` just ends up being calls to `weight`

https://github.com/Devsh-Graphics-Programming/Nabla/blob/124355408c8a96428205f9decb2c7eb6c75a4073/include/nbl/asset/filters/kernels/kernels.h#L87

P.S. We should probably make kernel tags (like pre-C++20 iterator tags) for separable, symmetric, etc.

Reply:
1. Is there something to do here or it can be resolved?
2. Should I remove the comment?

Reply:
what I meant is, the kernel weights **are not monochrome** so your LUT shouldn't be either.

Reply:
Removed the comment and the LUT not monochrome in #343.
Resolved.